### DISCUSSION TITLE
axum-extra cannot compile with cookie-private feature

### DISCUSSION THREAD

**Discussion by bbstilson** [tokio-rs/axum#1910] [Q&A] [answered]
- [x] I have looked for existing issues (including closed) about this

## Bug Report

In a new project  (`cargo new --bin foo`), I add the `axum-extra` dependency with the `cookie-private` flag, then run `cargo build`:

```
✗ cargo build
   Compiling axum-extra v0.7.2
error[E0271]: expected `FlatMap<FilterMap<ValueIter<'_, HeaderValue>, [closure@mod.rs:107:21]>, Split<'_, char>, [closure@mod.rs:108:19]>` to be an iterator that yields `Cookie<'static>`, but it yields `&str`
   --> /home/bbstilson/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-extra-0.7.2/src/extract/cookie/mod.rs:103:49
    |
103 | fn cookies_from_request(headers: &HeaderMap) -> impl Iterator<Item = Cookie<'static>> + '_ {
    |                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&str`, found struct `Cookie`

For more information about this error, try `rustc --explain E0271`.
error: could not compile `axum-extra` due to previous error
```

### Version

```
rustc --version
rustc 1.68.2 (9eb3afe9e 2023-03-27)
```

```
✗ cargo tree | grep axum
└── axum-extra v0.7.2
    ├── axum v0.6.12
    │   ├── axum-core v0.3.3
```

### Platform

```
uname -a
Linux pop-os 6.2.6-76060206-generic #202303130630~1679424972~22.04~4a8cde1 SMP PREEMPT_DYNAMIC Tue M x86_64 x86_64 x86_64 GNU/Linux
```

### Crates

```toml
[dependencies]
axum-extra = { version = "0.7.2", features = ["cookie-private"] }
```

**Comment by davidpdrsn** [comment, marked as answer]
Is the issue with `axum-request` or `axum-extra`? We don't maintain `axum-request` so if the issue is there try and ask in their repo. I tried making a new binary with

```toml
[dependencies]
axum-extra = { version = "0.7.2", features = ["cookie-private"] }
```

and that worked fine.

**Reply by bbstilson** [reply]
Sorry, I typo'd in my description, but the issue was with `axum-extra` as the only dependency in my `Cargo.toml`. You can see the compilation error is about `axum-extra`.  I mentioned in discord, though, that I don't think it's _really_ about `axum-extra`. I tested on my work computer and it does indeed compile. 

Maybe this is some cargo issue.

**Reply by bbstilson** [reply]
Okay I uninstalled everything (`rm -rf ~/.cargo`), reinstalled everything, and was able to compile this project. 

My guess is this was a cache issue. I have `RUSTC_WRAPPER="sccache"`, so I'm thinking the cache was poisoned.

Sorry for the noise. I was out of ideas 😖
